### PR TITLE
[model] Node2vec

### PR DESCRIPTION
**Pull request by HeuTab**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

File: gammagl/transforms/__init__.py [hunk 1]
```diff
@@ -13,6 +14,7 @@
     'SIGN',
     'NormalizeFeatures',
     'DropEdge',
+    'RandomWalk',
```
Comment (Zhanghyi) [inline gammagl/transforms/__init__.py:17]: 这个地方我感觉不适合放在transform里面，更偏向于sample

File: examples/node2vec/node2vec_trainer.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import argparse
+import tensorlayerx as tlx
+from gammagl.datasets import Planetoid
+from gammagl.models import Node2vecModel
+from gammagl.utils import calc_gcn_norm, mask_to_index
+from tensorlayerx.model import TrainOneStep, WithLoss
+from sklearn.linear_model import LogisticRegression
+
+
+class SemiSpvzLoss(WithLoss):
```
[the remark this answers is not in the document]
Comment (HeuTab) [reply examples/node2vec/node2vec_trainer.py:18]: 嗯对，应该是无监督损失

File: gammagl/models/__init__.py [hunk 3]
```diff
@@ -59,10 +60,11 @@
     'HardGATModel',
     'GPRGNNModel',
     'FILMModel',
+    'GEstimationN',
```
[the remark this answers is not in the document]
Comment (HeuTab) [reply gammagl/models/__init__.py:63]: OK.

**Comment by dddg617** [comment]
Refer to https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/models/test_node2vec.py and add tests.

**Pull request merged**